waspc/tests/Generator/Valid/PackageJsonTest.hs

```diff
@@ -0,0 +1,223 @@
+module Generator.Valid.PackageJsonTest (spec_PackageJson) where
+
+import qualified Data.Map as M
+import Test.Hspec
+import qualified Wasp.ExternalConfig.Npm.PackageJson as P
+import Wasp.Generator.Valid.PackageJson
+  ( DependencyType (..),
+    inDependency,
+    makeOptionalDepValidator,
+    makeRequiredDepValidator,
+  )
+import qualified Wasp.Generator.Valid.Validator as V
+
+spec_PackageJson :: Spec
+spec_PackageJson = do
+  describe "makeOptionalDepValidator" $ do
+    it "succeeds when optional dependency is not present" $ do
+      let pkgJson =
+            makePackageJson
+              []
+              []
+
+      let validator = makeOptionalDepValidator Runtime ("optional-pkg", "1.0.0")
+      V.execValidator validator pkgJson `shouldBe` []
```

Comment:
Ok a bit of fun, but I do want to push us all to try to make tests easy to read. These tests are quite boilerplatish, if this was normal code, one would be tempted to avoid it, so maybe let's also do it here, both for reducing duplication, but also for increasing readabilty, it is much easier to scan and understand test when you don't have to parse through all the boilerplate. Sometimes boilerplate is ok of course and it is not worth it trying to remove it, but I think there is some space here.

Can we make some nice DSL, for example something like this:
```suggestion
    it "succeeds when optional dependency is not present" $ do
      testSuccess
        MockPackageJson{ deps = [], devDeps = [] }
        (Runtime, ("optional-pkg", "1.0.0"))
```